main.tf

```diff
@@ -50,23 +49,39 @@ module "project-services" {
 
 resource "google_service_account" "runsa" {
   project      = var.project_id
-  account_id   = "${var.deployment_name}-run-sa"
+  account_id   = var.deployment_name
   display_name = "Service Account for Cloud Run"
 }
 
+# Handle Permissions
+variable "run_roles_list" {
+  description = "The list of roles that run needs"
+  type        = list(string)
+  default = [
+    "roles/secretmanager.secretAccessor",
+    "roles/cloudsql.instanceUser",
+    "roles/cloudsql.client",
+  ]
+}
+
 resource "google_project_iam_member" "allrun" {
```

Comment:
Rather than granting roles/secretmanager.secretAccessor could we grant granular secret level IAM so the SA does not have access to all secrets within the project? I have proposed we remove secret manager all together so please ignore if that is feasible?